props/saber_fett263_buttons.h

```diff
@@ -293,39 +659,745 @@ class SaberFett263Buttons : public virtual PropBase {
+          strcpy(value, "lockup ");
+          itoa(saved_choreography.clash_rec[i].sound_number1 + SFX_bgnlock.get_min_file(), value + strlen(value), 10);
+          strcat(value, " ");
+          itoa(saved_choreography.clash_rec[i].sound_number2 + file, value + strlen(value), 10);
+          strcat(value, " ");
+          itoa(saved_choreography.clash_rec[i].sound_number3 + SFX_endlock.get_min_file(), value + strlen(value), 10);
+          strcat(value, "\n");
+          out.Write(value);
+          break;
+        case SavedRehearsal::STANCE_END:
+        default:
+          break;
+      }
+    }
+    out.Write("end\n");
+    out.Close();
+    LOCK_SD(false);
+  }
+
+  void SaveChoreo() {
+    STDOUT.println("Saving Choreography");
+    WriteChoreo("choreo.tmp");
+    WriteChoreo("choreo.ini");
+  }
+#endif
+
+  void GenerateIniFiles() {
+    SaveState(current_preset_.preset_num);
+    SaveGestureState();
+  }
+
+#ifdef FETT263_EDIT_MODE_MENU
+  // Color / Style Editing
+  HSL hsl_;
+  ShowColorSingleBladeTemplate<RotateColorsX<Variation,ShowColorStyle>> show_color_;
+  ShowColorSingleBladeTemplate<Mix<Bump<Int<16384>,Int<14000>>,RotateColorsX<Variation,RgbArg<1,Rgb<255,0,0>>>,ShowColorStyle>> bump_color_;
+  ShowColorSingleBladeTemplate<Mix<SmoothStep<Int<26000>,Int<8000>>,RotateColorsX<Variation,RgbArg<1,Rgb<255,0,0>>>,ShowColorStyle>> tip_color_;
+  ShowColorSingleBladeTemplate<Mix<SmoothStep<Int<7000>,Int<-6000>>,RotateColorsX<Variation,RgbArg<1,Rgb<255,0,0>>>,ShowColorStyle>> hilt_color_;
+  ShowColorSingleBladeTemplate<Mix<SmoothStep<Int<7000>,Int<-6000>>,Black,RotateColorsX<Variation,ShowColorStyle>>> pre_color_;
+  ShowColorSingleBladeTemplate<LengthEdit<RotateColorsX<Variation,ShowColorStyle>,Pulsing<White,Rgb<100,100,100>,800>>> show_length_;
+  ShowColorSingleBladeTemplate<Mix<Bump<IntEdit,Int<10000>>,Black,ShowColorStyle>> show_clash_location_;
+  ShowColorSingleBladeTemplate<Mix<SmoothStep<IntEdit,Int<6000>>,Black,ShowColorStyle>> show_drag_size_;
+  ShowColorSingleBladeTemplate<Mix<SmoothStep<IntEdit,Int<-6000>>,Black,ShowColorStyle>> show_emitter_size_;
+#if (NUM_BLADES > 1)
+  ShowColorSingleBladeTemplate<Pulsing<RotateColorsX<Variation,ShowColorStyle>,Black,800>> show_preview_;
+#endif
+
+  Color16 GetColorArg(int blade, int arg) {
+    char argspace[32];
+    if (style_parser.GetArgument(current_preset_.GetStyle(blade), arg + 2, argspace)) {
+      char* tmp;
+      int r = strtol(argspace, &tmp, 0);
+      int g = strtol(tmp+1, &tmp, 0);
+      int b = strtol(tmp+1, NULL, 0);
+      return Color16(r,g,b);
+    }
+    return Color16();
+  }
+
+  // Show Full Blade Preview
+  void ShowFull() {
+    show_color_.Start(blade_num_);
+    ShowColorStyle::SetColor(GetColorArg(blade_num_, effect_num_)); 
+    saved_color_ = GetColorArg(blade_num_, effect_num_);
+    hsl_ = saved_color_.toHSL();
+    hsl_angle_ = fusor.angle2();
+  }
+
+  // Show Partial Blade Preview (Clash, Lockup, Drag, Melt, Preon, Emitter)
+  void ShowPart() {
```

Comment:
This function is used to enter a color editing menu.
It would be nice if the comment said that. (Same for ShowFull())

Reply:
comment updated